# Import Dataset

In [1]:
import pandas as pd

In [2]:
df = pd.read_csv('https://raw.githubusercontent.com/taufikyasirs/marketing_campaign_marketektive/master/marketing_campaign.csv',delimiter=';')

In [3]:
df.head()

ID  Year_Birth   Education Marital_Status   Income  Kidhome  Teenhome  \
0  5524        1957  Graduation         Single  58138.0        0         0   
1  2174        1954  Graduation         Single  46344.0        1         1   
2  4141        1965  Graduation       Together  71613.0        0         0   
3  6182        1984  Graduation       Together  26646.0        1         0   
4  5324        1981         PhD        Married  58293.0        1         0   

  Dt_Customer  Recency  MntWines  ...  NumWebVisitsMonth  AcceptedCmp3  \
0  2012-09-04       58       635  ...                  7             0   
1  2014-03-08       38        11  ...                  5             0   
2  2013-08-21       26       426  ...                  4             0   
3  2014-02-10       26        11  ...                  6             0   
4  2014-01-19       94       173  ...                  5             0   

   AcceptedCmp4  AcceptedCmp5  AcceptedCmp1  AcceptedCmp2  Complain  \
0             0             0             0             0         0   
1             0             0             0             0         0   
2             0             0             0             0         0   
3             0             0             0             0         0   
4             0             0             0             0         0   

   Z_CostContact  Z_Revenue  Response  
0              3         11         1  
1              3         11         0  
2              3         11         0  
3              3         11         0  
4              3         11         0  

[5 rows x 29 columns]

# Descriptive Statistics

In [4]:
#nama kolom yang ada pada dataset
df.columns.to_list()

['ID',
 'Year_Birth',
 'Education',
 'Marital_Status',
 'Income',
 'Kidhome',
 'Teenhome',
 'Dt_Customer',
 'Recency',
 'MntWines',
 'MntFruits',
 'MntMeatProducts',
 'MntFishProducts',
 'MntSweetProducts',
 'MntGoldProds',
 'NumDealsPurchases',
 'NumWebPurchases',
 'NumCatalogPurchases',
 'NumStorePurchases',
 'NumWebVisitsMonth',
 'AcceptedCmp3',
 'AcceptedCmp4',
 'AcceptedCmp5',
 'AcceptedCmp1',
 'AcceptedCmp2',
 'Complain',
 'Z_CostContact',
 'Z_Revenue',
 'Response']

In [5]:
print(f'Terdapat {df.shape[0]} baris data dan {df.shape[1]} kolom didalam dataset')

Terdapat 2240 baris data dan 29 kolom didalam dataset


In [6]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2240 entries, 0 to 2239
Data columns (total 29 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   ID                   2240 non-null   int64  
 1   Year_Birth           2240 non-null   int64  
 2   Education            2240 non-null   object 
 3   Marital_Status       2240 non-null   object 
 4   Income               2216 non-null   float64
 5   Kidhome              2240 non-null   int64  
 6   Teenhome             2240 non-null   int64  
 7   Dt_Customer          2240 non-null   object 
 8   Recency              2240 non-null   int64  
 9   MntWines             2240 non-null   int64  
 10  MntFruits            2240 non-null   int64  
 11  MntMeatProducts      2240 non-null   int64  
 12  MntFishProducts      2240 non-null   int64  
 13  MntSweetProducts     2240 non-null   int64  
 14  MntGoldProds         2240 non-null   int64  
 15  NumDealsPurchases    2240 non-null   i

In [7]:
df.isna().sum()

ID                      0
Year_Birth              0
Education               0
Marital_Status          0
Income                 24
Kidhome                 0
Teenhome                0
Dt_Customer             0
Recency                 0
MntWines                0
MntFruits               0
MntMeatProducts         0
MntFishProducts         0
MntSweetProducts        0
MntGoldProds            0
NumDealsPurchases       0
NumWebPurchases         0
NumCatalogPurchases     0
NumStorePurchases       0
NumWebVisitsMonth       0
AcceptedCmp3            0
AcceptedCmp4            0
AcceptedCmp5            0
AcceptedCmp1            0
AcceptedCmp2            0
Complain                0
Z_CostContact           0
Z_Revenue               0
Response                0
dtype: int64

- Dataset terdiri dari **29 columns** dan **2240 rows** data
- Variasi tipe data yang ada pada dataset adalah **int64, object, float64**
- Pada kolom **Income** memiliki **2216 rows nilai non-null** dari keseluruhan **2240 rows** yang berarti terdapat **24 nilai null / missing values** sedangkan untuk kolom lainya jumlah rows non-null sama dengan keseluruan rows data yang ada
- Type data keseluruhan sudah tidak mengalami masalah hanya sebaiknya pada data ID digunakan type string saja karena ID merupakan identitas unik pelanggan yang tentunya tidak relevan untuk dilakukan aritmatika, kemudian untuk data Dt_customer yang isinya berkaitan dengan waktu dan tanggal ada baiknya dijadikan ke tipe Datetime saja

In [8]:
df['Dt_Customer'] = pd.to_datetime(df['Dt_Customer'])
df['ID']= df['ID'].astype(str)

In [9]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2240 entries, 0 to 2239
Data columns (total 29 columns):
 #   Column               Non-Null Count  Dtype         
---  ------               --------------  -----         
 0   ID                   2240 non-null   object        
 1   Year_Birth           2240 non-null   int64         
 2   Education            2240 non-null   object        
 3   Marital_Status       2240 non-null   object        
 4   Income               2216 non-null   float64       
 5   Kidhome              2240 non-null   int64         
 6   Teenhome             2240 non-null   int64         
 7   Dt_Customer          2240 non-null   datetime64[ns]
 8   Recency              2240 non-null   int64         
 9   MntWines             2240 non-null   int64         
 10  MntFruits            2240 non-null   int64         
 11  MntMeatProducts      2240 non-null   int64         
 12  MntFishProducts      2240 non-null   int64         
 13  MntSweetProducts     2240 non-nul

In [10]:
round(df.describe(),2)

Year_Birth     Income  Kidhome  Teenhome  Recency  MntWines  MntFruits  \
count     2240.00    2216.00  2240.00   2240.00  2240.00   2240.00    2240.00   
mean      1968.81   52247.25     0.44      0.51    49.11    303.94      26.30   
std         11.98   25173.08     0.54      0.54    28.96    336.60      39.77   
min       1893.00    1730.00     0.00      0.00     0.00      0.00       0.00   
25%       1959.00   35303.00     0.00      0.00    24.00     23.75       1.00   
50%       1970.00   51381.50     0.00      0.00    49.00    173.50       8.00   
75%       1977.00   68522.00     1.00      1.00    74.00    504.25      33.00   
max       1996.00  666666.00     2.00      2.00    99.00   1493.00     199.00   

       MntMeatProducts  MntFishProducts  MntSweetProducts  ...  \
count          2240.00          2240.00           2240.00  ...   
mean            166.95            37.53             27.06  ...   
std             225.72            54.63             41.28  ...   
min               0.00             0.00              0.00  ...   
25%              16.00             3.00              1.00  ...   
50%              67.00            12.00              8.00  ...   
75%             232.00            50.00             33.00  ...   
max            1725.00           259.00            263.00  ...   

       NumWebVisitsMonth  AcceptedCmp3  AcceptedCmp4  AcceptedCmp5  \
count            2240.00       2240.00       2240.00       2240.00   
mean                5.32          0.07          0.07          0.07   
std                 2.43          0.26          0.26          0.26   
min                 0.00          0.00          0.00          0.00   
25%                 3.00          0.00          0.00          0.00   
50%                 6.00          0.00          0.00          0.00   
75%                 7.00          0.00          0.00          0.00   
max                20.00          1.00          1.00          1.00   

       AcceptedCmp1  AcceptedCmp2  Complain  Z_CostContact  Z_Revenue  \
count       2240.00       2240.00   2240.00         2240.0     2240.0   
mean           0.06          0.01      0.01            3.0       11.0   
std            0.25          0.11      0.10            0.0        0.0   
min            0.00          0.00      0.00            3.0       11.0   
25%            0.00          0.00      0.00            3.0       11.0   
50%            0.00          0.00      0.00            3.0       11.0   
75%            0.00          0.00      0.00            3.0       11.0   
max            1.00          1.00      1.00            3.0       11.0   

       Response  
count   2240.00  
mean       0.15  
std        0.36  
min        0.00  
25%        0.00  
50%        0.00  
75%        0.00  
max        1.00  

[8 rows x 25 columns]

In [11]:
#memisahkan kolom kategorik dan numerik
numerik = ['Year_Birth'
           ,'Income'
           ,'Kidhome'
           ,'Teenhome'
           ,'Recency'
           ,'MntWines'
           ,'MntFruits'
           ,'MntMeatProducts'
           ,'MntFishProducts'
           ,'MntSweetProducts'
           ,'MntGoldProds'
           ,'NumDealsPurchases'
           ,'NumWebPurchases'
           ,'NumCatalogPurchases'
           ,'NumStorePurchases'
           ,'NumWebVisitsMonth'
           ,'Z_CostContact'
           ,'Z_Revenue']

kategorik = ['ID'
             ,'Education'
             ,'Marital_Status'
             ,'AcceptedCmp1'
             ,'AcceptedCmp2'
             ,'AcceptedCmp3'
             ,'AcceptedCmp4'
             ,'AcceptedCmp5'
             ,'Complain'
             ,'Response'
             ,'Dt_Customer']

In [12]:
round(df[numerik].describe(),2)

Year_Birth     Income  Kidhome  Teenhome  Recency  MntWines  MntFruits  \
count     2240.00    2216.00  2240.00   2240.00  2240.00   2240.00    2240.00   
mean      1968.81   52247.25     0.44      0.51    49.11    303.94      26.30   
std         11.98   25173.08     0.54      0.54    28.96    336.60      39.77   
min       1893.00    1730.00     0.00      0.00     0.00      0.00       0.00   
25%       1959.00   35303.00     0.00      0.00    24.00     23.75       1.00   
50%       1970.00   51381.50     0.00      0.00    49.00    173.50       8.00   
75%       1977.00   68522.00     1.00      1.00    74.00    504.25      33.00   
max       1996.00  666666.00     2.00      2.00    99.00   1493.00     199.00   

       MntMeatProducts  MntFishProducts  MntSweetProducts  MntGoldProds  \
count          2240.00          2240.00           2240.00       2240.00   
mean            166.95            37.53             27.06         44.02   
std             225.72            54.63             41.28         52.17   
min               0.00             0.00              0.00          0.00   
25%              16.00             3.00              1.00          9.00   
50%              67.00            12.00              8.00         24.00   
75%             232.00            50.00             33.00         56.00   
max            1725.00           259.00            263.00        362.00   

       NumDealsPurchases  NumWebPurchases  NumCatalogPurchases  \
count            2240.00          2240.00              2240.00   
mean                2.33             4.08                 2.66   
std                 1.93             2.78                 2.92   
min                 0.00             0.00                 0.00   
25%                 1.00             2.00                 0.00   
50%                 2.00             4.00                 2.00   
75%                 3.00             6.00                 4.00   
max                15.00            27.00                28.00   

       NumStorePurchases  NumWebVisitsMonth  Z_CostContact  Z_Revenue  
count            2240.00            2240.00         2240.0     2240.0  
mean                5.79               5.32            3.0       11.0  
std                 3.25               2.43            0.0        0.0  
min                 0.00               0.00            3.0       11.0  
25%                 3.00               3.00            3.0       11.0  
50%                 5.00               6.00            3.0       11.0  
75%                 8.00               7.00            3.0       11.0  
max                13.00              20.00            3.0       11.0

**Year_Birth:**<br>
- Tahun kelahiran tertua (min) yaitu **1893**, hal ini agak janggal saja karena jika dibandingkan dengan kelahiran termuda yang ada pada data terdapat selisih yang hampir 100 tahun, perlu dikonfirmasi apakah data ini benar atau terjadi kesalahan input

**Income**<br>
- Nilai rata-rata nya adalah 52247.25 dan Nilai tengah(Median) nya adalah 51381.50,nilai rata-rata lebih besar dari pada median maka dari kedua nilai tersebut secara tidak langsung dapat dikatakan bahwa ada kemencengan distribusi data

**Kidhome dan Teenhome**<br>
- Dari data ini tidak menunjukan keanehan

**Recency**<br>
- Nilai rata-rata dan median data ini hampir sama sehingga dapat dikatakan kemungkinan besar datanya berdistribusi normal

**data produk bulanan pelanggan**<br>
1. MntWines, memiliki nilai mean = 303.94 dan median = 173.5
2. MntFruits, memiliki nilai mean = 26.30 dan median = 8.00
3. MntMeatProducts, memiliki nilai mean = 166.95 dan median = 67.00
4. MntFishProducts, memiliki nilai mean = 37.53 dan median = 12.00
5. MntSweetProducts, memiliki nilai mean = 27.06 dan median = 8.00
6. MntGoldProds, memiliki nilai mean = 44.02 dan median = 24.00

- karena terdapat perbedaan nilai yang lumayan signifikan dari kedua nilai tersebut secara tidak langsung dapat dikatakan bahwa ada kemencengan distribusi data

**data pembelian**
1. NumDealsPurchases memiliki nilai mean = 2.33 dan median = 2.00
2. NumWebPurchases memiliki nilai mean = 4.08 dan median = 4.0
3. NumCatalogPurchases memiliki nilai mean = 2.66 dan median = 2.00
4. NumStorePurchases memiliki nilai mean = 5.79 dan median = 5.00

- karena terdapat perbedaan nilai yang relatif tidak besar dari kedua nilai tersebut secara tidak langsung dapat dikatakan bahwa distribusi data mendekati normal

**NumWebVisitsMont**
- memiliki nilai mean = 5.32 dan median = 6.00 karena terdapat perbedaan nilai yang relatif tidak besar dari kedua nilai tersebut secara tidak langsung dapat dikatakan bahwa distribusi data mendekati normal

**untuk nilai Z_CostContact dan Z_Revenue dengan standar deviasi adalah 0 dan nilai lainya yang sama menunjukan bahwa data ini memiliki 1 nilai yang sama pada setiap row nya**

In [13]:
cat_des = df[kategorik].astype("category").describe().transpose()
cat_des['top_percentage'] = cat_des['freq']/cat_des['count']*100
cat_des

count unique                  top  freq top_percentage
ID              2240   2240                    0     1       0.044643
Education       2240      5           Graduation  1127        50.3125
Marital_Status  2240      8              Married   864      38.571429
AcceptedCmp1    2240      2                    0  2096      93.571429
AcceptedCmp2    2240      2                    0  2210      98.660714
AcceptedCmp3    2240      2                    0  2077      92.723214
AcceptedCmp4    2240      2                    0  2073      92.544643
AcceptedCmp5    2240      2                    0  2077      92.723214
Complain        2240      2                    0  2219        99.0625
Response        2240      2                    0  1906      85.089286
Dt_Customer     2240    663  2012-08-31 00:00:00    12       0.535714

- Pada data ID semuanya hanya muncul sekali ditunjukkan oleh count dan unique yang nilainya sama, karena itu dapat dinyatakan bahwa tidak ada duplicate ID pada data
- mayoritas pelangan adalah pada kategori pendidikan `Graduation` sebanyak 1127 orang, nilai ini sudah setengah dari keseluruhan data
- untuk status respon campaign(1-5) mayoritas pelanggan tidak merespon (diatas 90 % pelanggan tidak merespon)
- jumlah complain relatif kecil hanya 0.04%
- dan respon customer relatif rendah dilihat dari persentase customer yang tidak memberikan respon adalah 85.09%